tensorflow/compiler/mlir/xla/tests/lhlo-legalize-to-affine.mlir

```diff
@@ -143,3 +143,22 @@ func @int_sub_op(%lhs: memref<7xi32>, %rhs: memref<7xi32>,
       : (memref<7xi32>, memref<7xi32>, memref<7xi32>) -> ()
   return
 }
+
+// Dot tests.
+// CHECK-LABEL: func @float_dot_op
+func @float_dot_op(%lhs: memref<memref<7x3xf32>, %rhs: 
+                  memref<3x4xf32>, %result: memref<7x4xf32> ) -> () {
+    // CHECK-NEXT: affine.for %[[I:.*]] = 0 to 7 {
+    // CHECK-NEXT: affine.for %[[J:.*]] = 0 to 3 {
+    // CHECK-NEXT: affine.for %[[K:.*]] = 0 to 4 {
+    // CHECK-NEXT: %[[LHS:.*]] = affine.load %{{.*}}[%[[I]], %[[J]]] : memref<7x3xf32>
+    // CHECK-NEXT: %[[RHS:.*]] = affine.load %{{.*}}[%[[J]], %[[K]]] : memref<3x4xf32>
+    // CHECK-NEXT: %[[RESULT:.*]] = affine.load %{{.*}}[%[[I]], %[[K]]] : memref<7x4xf32>
+    // CHECK-NEXT: %[[MULT:.*]] = mulf %[[LHS]], %[[RHS]] : f32
+    // CHECK-NEXT: %[[ADD:.*]] =  addf %[[MULT]], %[[RESULT]] : f32
+    // CHECK-NEXT: affine.store %[[ADD]], %{{.*}}[%[[I]], %[[K]]] : memref<7x4xf32> 
```

Comment:
This looks fine except that you want to generate a different loop order: so that it's LHS I, K,  RHS K, J, and RESULT I, J.